src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -76,22 +98,26 @@ public final void publishWithExpiry(final Message message, final long expiryInMs
                     .headers(ImmutableMap.of(MESSAGE_EXPIRY_TEXT, expiresAt))
                     .build();
         }
-        publish(message, properties);
+        publish(message, properties, RMQOperation.PUBLISH_WITH_EXPIRY);
     }
 
     public final void publish(final Message message) throws Exception {
-        publish(message, MessageProperties.MINIMAL_PERSISTENT_BASIC);
+        publish(message, MessageProperties.MINIMAL_PERSISTENT_BASIC, RMQOperation.PUBLISH);
     }
 
-    public final void publish(final Message message, final AMQP.BasicProperties properties) throws Exception {
-        String routingKey;
-        if (config.isSharded()) {
-            routingKey = NamingUtils.getShardedQueueName(queueName, getShardId());
-        } else {
-            routingKey = queueName;
-        }
-        val enrichedProperties = getEnrichedProperties(properties);
-        publishChannel.basicPublish(config.getExchange(), routingKey, enrichedProperties, mapper().writeValueAsBytes(message));
+    public final void publish(final Message message, final AMQP.BasicProperties properties, final RMQOperation operation) throws Exception {
+        String routingKey = getRoutingKey();
```

Comment:
routing key should only be fetched if sharding is enabled for the queue

Reply:
getRoutingKey method checks `config.isSharded()` and returns sharded queue name or queue name accordingly.